templates/.snapshots/TestDevspaceYaml-devspace.yaml.tpl-devspace.yaml.snapshot

```diff
@@ -363,6 +373,48 @@ profiles:
             name: E2E
             value: "true"
 
+  - name: binarySync
+    description: Synchronizes just content of bin folder and don't do any build related stuff in the devspace pod
+    activation:
+      - vars:
+          DEVENV_SYNC_BINARIES: "true"
+    patches:
+      - op: replace
+        path: dev.app.devImage
+        value: gcr.io/outreach-docker/bootstrap/dev-slim:stable
+      - op: replace
+        path: dev.app.sync
+        value:
+          - path: ./bin:${DEV_CONTAINER_WORKDIR}
+            printLogs: true
+            disableDownload: true
+            waitInitialSync: true
+      - op: add
+        path: dev.app.patches
+        value: 
+          op: replace
+          path: spec.containers[0].resources.limits
+          value:
+            cpu: 200m
+            memory: 400Mi
```

Comment:
We might just want to make this the default request instead, since that's really the only time we want to replace these (I imagine we want to leave the default limits for services that HAVE bumped them up)

Reply:
Will it be enough for devspace build mode? It feels it needs gigabites to do all the building / linting stuff. That why I would keep it different for bin mode, where 400MI is enough and out of this PR

Reply:
This limit applies to binary sync, right? I'm more thinking that we need to change the default values so they persist because if we don't we're going to limit all of these applications to 200m/400Mi _while_ they are running (since we need it for the build, but technically after build we want to run with whatever limits they had specified, which seems harder to implement). Lemme know if that's not clear!

Reply:
Currently, devspace compile mode does not alter resource limits so it goes with what the service has in production. I raised [concern](https://outreach-hq.slack.com/archives/C04M2RNHY93/p1686809080876359) about that. Originally that resource limit was removed for devspace and it was eating significant resources and making devenv starting for resources. 
Devspace in bin mode (this PR) is changing default resource limits to 200m/400Mi to be able to work with its use cases -> delve, hot reloading.

Reply:
We can either:

- Remove limit for devspace compile mode - devspace will work, other pods might be struggling
- Create some limit for devspace compile mode - Like 2CPU and 2GB ram. Maybe more.

I would rather keep that change out of this PR, since I'm touching this part just for devspace binary mode.

Reply:
I changed these lines to overwrite whole resources section. Previous approach was not working when some teams defined `resource.request` for devenv that were above the current limits for binary sync. I'm still thinking for better solution, but still I think this is probably good enough for now.

Reply:
This part is removed @jaredallard